kernel/dosfns.c

```diff
@@ -271,6 +319,10 @@ COUNT SftSeek(int sft_idx, LONG new_pos, unsigned mode)
   {
     new_pos = 0;
   }
+  else if (try_long_compat_remote_seek(s, new_pos, mode, p_result))
+  {
+    return SUCCESS;
+  }
   else if (mode == SEEK_CUR)
   {
```

Comment:
Any reason you haven't made it
specific to SEEK_CUR?

Reply:
I think it is cleaner this way. And we don't depend on the redirector support for the 32-bit seek from EOF service. (I'm not certain whether that is correct in mfs now.) Also, for some corner cases it is better to always call into 1142h. Eg, what if the 32-bit seek ends up with sft_position = FFFFffffh?